compiler/qsc_formatter/src/formatter.rs

```diff
@@ -490,10 +526,10 @@ impl<'a> Formatter<'a> {
         }
     }
 
-    /// Updates the type_param_state of the FormatterState based
+    /// Updates the type_param_state and import_export_state of the FormatterState based
     /// on the left and right token kinds. Returns the delimiter
     /// state of the left and right tokens.
-    fn update_type_param_state(
+    fn update_formatter_state(
```

Comment:
We talked about this change a bit in-person, but looking at it now I really don't think this function should handle the update to both of the state fields. Most of the logic in this function still pertains to `TypeParameterListState` and the match statement that you introduced tries to handle both states simultaneously, which is confusing and might be error prone as we maintain this going forward.
From a naming perspective, I would expect `update_formatter_state` to handle updating all of the state of the formatter, but it just handles two of the state fields, which is a bit awkward. Though, I'm not suggesting to merge in the logic to update `indent_level`, `delim_newlines_stack`, and `spec_decl_state` fields also, as I think that function would be way too complex.

My suggestion here is to move the calls to `Delimiter::from_concrete_token_kind` (I just realized that's misspelled in the code) to outside of this function, have this function just updated the type_param_state as before, add a new function that just updates the import_export state:

```qsharp
fn update_import_export_state(&mut self, left_kind: &ConcreteTokenKind) {
    use qsc_frontend::keyword::Keyword;
    use ConcreteTokenKind::*;
    use TokenKind::*;
    match left_kind {
        Syntax(Semi) => {
            self.import_export_state = ImportExportState::NoState;
        }
        Syntax(Keyword(Keyword::Import | Keyword::Export)) => {
            self.import_export_state = ImportExportState::HandlingImportExportStatement;
        }
        _ => (),
    }
}
```

Reply:
Thanks for making that change!